### ISSUE TITLE
Javascript error on uploading an extension

### ISSUE THREAD

**Issue by andymckay** [mozilla/addons-linter#1166]
1. Packaged my side tabs extension: https://github.com/andymckay/sidebar-tabs 
2. Uploaded to AMO.
3. Got a validation error:

https://addons.mozilla.org/en-US/developers/upload/0ae81bce24f1445793262a5b3a4194d1

4.  The error is on the catch here:

```
   try {
      pos = sidetabs.getPos(element.id) + 1;
    } catch (e if e instanceof TypeError) {
      console.log(`No tab of id ${element.id} for drop, invalid target?`);
    }
```

I can work around that, or just remove the check, but it would be nice if it could be uploaded since that's valid JS for Firefox I believe based on: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch

**Comment by EnTeQuAk** [comment]
``` Non-standard
This feature is non-standard and is not on a standards track. Do not use it on production sites facing the Web: it will not work for every user. There may also be large incompatibilities between implementations and the behavior may change in the future.
```

If that is something that's needed it should be reported against https://github.com/eslint/espree/issues to add support for it.

Closing this for now since it's good enough documented to better not use conditional catch statements.
